### PR TITLE
Added configuration documentation.

### PR DESCRIPTION
**Pull request by tobiasake**
EI Frontend was missing the configuration documentation.
This pull-request adds missing documentations about configuration of EI Frontend.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: wiki/markdown/configuration.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Configuration
+
+To be able to execute and use Eiffel-Intelligence Front-end, some configuration need to provided.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: Here front-end is hyphened but with first capital letter, later no hyphen is used etc.. Please change all occurrences to use "front-end" to be consistent with other documentation.

File: wiki/markdown/configuration.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+All configurations in made via application.properties file. These properties can also be provided as Java properties arguments as well.
+
+## Eiffel-Intelligence Backend
+
+Eiffel-Intelligence Frontend is dependent that at least one EI Backend instance is up and running.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: Same comment with backend here, add hyphen and lowercase the word: "back-end"

File: wiki/markdown/configuration.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Configuration
+
+To be able to execute and use Eiffel-Intelligence Front-end, some configuration need to provided.
+All configurations in made via application.properties file. These properties can also be provided as Java properties arguments as well.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: Should we maybe have examples of a command with some Java properties following this sentence?

File: wiki/markdown/configuration.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+## Eiffel-Intelligence Backend
+
+Eiffel-Intelligence Frontend is dependent that at least one EI Backend instance is up and running.
+It is possible to configure several EI Backend instances.
+
+EI Backebnd instances is configured by setting "ei.backend.instances.list.json.content" property, example:
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: back-end misspelled. Highlight the property name in bold to make it more visible.

File: wiki/markdown/configuration.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    ei.backend.instances.list.json.content=[{ "contextPath": "", "port": "8090", "name": "EI-Backend-1", "host": "ei-backend-host-1", "https": false, "defaultBackend": true},\
+                                            { "contextPath": "", "port": "8090", "name": "EI-Backend-2", "host": "ei-backend-host-2", "https": false, "defaultBackend": false},\
+                                            { "contextPath": "", "port": "8090", "name": "EI-Backend-3", "host": "ei-backend-host-3", "https": false, "defaultBackend": false},\
+                                            { "contextPath": "", "port": "8090", "name": "EI-Backend-4", "host": "ei-backend-host-4", "https": false, "defaultBackend": false}]
+
+You can set as many EI Backend instances as you intend to use.
+
+## Eiffel-Intelligence Frontend server settings
+
+Which port EIffel-Intelligence Front-end web-server should be started with is set by property "server.port":
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: by the property...

File: wiki/markdown/configuration.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+"ei.frontend.context.path" property is used when EI Frontend applications is started with any context path in web-server.
+If running locally with "mvn spring-boot:run" no context-path is added which means you can leave the ei.frontend.context.path property empty.
+
+    ei.frontend.service.host=localhost
+    ei.frontend.service.port=${server.port}
+    ei.frontend.context.path=
+
+## HTTPS security
+
+If Eiffel-Intelligence Frontend need to be execute with secure HTTPS connections, then these properties need to be set:
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: need --> needs

File: wiki/markdown/configuration.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+    ei.use.secure.http.frontend=false
+    server.ssl.key-store: <keystore.p12>
+    server.ssl.key-store-password: <mypassword>
+    server.ssl.key-store-type: <PKCS12>
+    server.ssl.key-alias: <tomcat>
+
+Read more in Spring documentations about how to enable HTTPS security in Spring applications.
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: Add link to Spring documentation

File: wiki/markdown/configuration.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+## Customize documentation links
+
+It is possible to add and change Documentation url links that is seen in the Eiffel-Intelligence Front-end Web-UI.
+Documentaiton url links is configured by property "ei.eiffel.documentation.urls", example:
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: Documentaiton  --> Documentation

File: wiki/markdown/configuration.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+    ei.eiffel.documentation.urls={ "EI Frontend Documentation": "https://eiffel-community.github.io/eiffel-intelligence-frontend",\
+                               "EI Frontend GitHub": "https://github.com/eiffel-community/eiffel-intelligence-frontend",\
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: These references of back-end and front-end should also be consistent with lowercase and hyphened.

File: wiki/markdown/configuration.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+                               "EI Backend Documentation": "https://eiffel-community.github.io/eiffel-intelligence",\
+                               "EI Backend GitHub": "https://github.com/eiffel-community/eiffel-intelligence",\
+                               "Eiffel Github main page": "https://github.com/eiffel-community/eiffel",\
+                               "Test Rules User Guide": "https://github.com/eiffel-community/eiffel-intelligence-frontend/blob/master/wiki/markdown/test-rules.md" }
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/configuration.md]: Use lower case, and I think rephrasing it to "User guide for test rules page" sounds better

File: wiki/markdown/running-eiffel-intelligence-frontend.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
+# Running Eiffel Intelligence Front-end
+
+## _Prerequisites_
+
+Eiffel intelligence Front-end is a Spring microservice distributed in a war file. To start
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/running-eiffel-intelligence-frontend.md]: Same comments in this file, use lower case and hyphened spelling of back-end and front-end references. Also add a comma after "To start the service"

File: wiki/markdown/running-eiffel-intelligence-frontend.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+the service download the war file from
+[jitpack.io](https://jitpack.io/#eiffel-community/eiffel-intelligence-frontend) and look for the
+latest version. Now replace the latest version in the link below:
+
+    https://jitpack.io/com/github/eiffel-community/eiffel-intelligence-frontend/<version>/eiffel-intelligence-<version>.war
+
+If you want to test the latest code in github clone the project and compile it
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/running-eiffel-intelligence-frontend.md]: github --> GitHub

File: wiki/markdown/running-eiffel-intelligence-frontend.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+append **_-DskipTests_** if you want to skip the tests since the latest on
+master always has passed the tests. The war file should now be found under
+target folder in your cloned project.
+
+If you run from source code, you can run Eiffel-Intelligence Front-end with maven command:
+
+    mvn spring-boot:run
+
+Or you can run executable war file (located in target folder, if running from source code):
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/running-eiffel-intelligence-frontend.md]: Or you can run the executable

File: wiki/markdown/running-eiffel-intelligence-frontend.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+
+    java -jar eiffel-intelligence-frontend-<version>.war
+
+if you want to run with default configuration for local message buss and
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/running-eiffel-intelligence-frontend.md]: buss --> bus

File: wiki/markdown/running-eiffel-intelligence-frontend.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+
+Own configuration can be provided with
+
+    java -jar eiffel-intelligence-frontend-<version>.war --spring.config.location=file:<path to own application.properties>
+
+remember to keep the name of the properties file if you are a beginner to
+Spring. More advanced Spring user can look [here](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html)
+for more information about using external configuration.
+
+If only few properties need to be overridden then use java opts, for example
```
Comment (e-pettersson-ericsson) [inline wiki/markdown/running-eiffel-intelligence-frontend.md]: If only a few properties need to be overriden, then use Java opts ...

File: wiki/markdown/configuration.md (hunk 1 above)
Comment (tobiasake) [reply wiki/markdown/configuration.md]: fixed.

File: wiki/markdown/configuration.md (hunk 3 above)
Comment (tobiasake) [reply wiki/markdown/configuration.md]: There is some examples of how to do that on Running Eiffel Intelligence Frontend page.

File: wiki/markdown/configuration.md (hunk 4 above)
Comment (tobiasake) [reply wiki/markdown/configuration.md]: fixed.

File: wiki/markdown/configuration.md (hunk 5 above)
Comment (tobiasake) [reply wiki/markdown/configuration.md]: fixed.

File: wiki/markdown/running-eiffel-intelligence-frontend.md (hunk 11 above)
Comment (tobiasake) [reply wiki/markdown/running-eiffel-intelligence-frontend.md]: fixed.

File: wiki/markdown/running-eiffel-intelligence-frontend.md (hunk 13 above)
Comment (tobiasake) [reply wiki/markdown/running-eiffel-intelligence-frontend.md]: fixed.

File: wiki/markdown/running-eiffel-intelligence-frontend.md (hunk 14 above)
Comment (tobiasake) [reply wiki/markdown/running-eiffel-intelligence-frontend.md]: fixed,, removed that due to it is not valid for frontend. This file is based on same file from EI Backend, with some modifications for frontend usage point-of-view.

File: wiki/markdown/configuration.md (hunk 9 above)
Comment (tobiasake) [reply wiki/markdown/configuration.md]: fixed.

File: wiki/markdown/configuration.md (hunk 10 above)
Comment (tobiasake) [reply wiki/markdown/configuration.md]: fixed.

**Pull request merged**